### ISSUE TITLE
Incorrect warning when summarising empty data frame

### ISSUE THREAD

**Issue by hadley** [tidyverse/dplyr#6675]
``` r
library(dplyr, warn.conflicts = FALSE)

df <- tibble(x = integer())
df |> summarise(y = x + 1)
#> Warning: Returning more (or less) than 1 row per `summarise()` group was deprecated in
#> dplyr 1.1.0.
#> ℹ Please use `reframe()` instead.
#> ℹ When switching from `summarise()` to `reframe()`, remember that `reframe()`
#>   always returns an ungrouped data frame and adjust accordingly.
#> # A tibble: 0 × 1
#> # … with 1 variable: y <dbl>
```

<sup>Created on 2023-02-01 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by DavisVaughan** [comment]
I don't think this is a bug. You have 1 group and it returned a result of size 0, which is what we are trying to avoid.

If you go back to dplyr 0.8.5 before we allowed multi-row `summarise()`, it gives an error here, which is what we'd eventually get back to

``` r
library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union

df <- tibble(x = integer())

summarise(df, y = x + 1)
#> Error: Column `y` must be length 1 (a summary value), not 0
```

<sup>Created on 2023-02-01 with [reprex v2.0.2.9000](https://reprex.tidyverse.org)</sup>

**Comment by hadley** [comment]
Oh yeah, good point.

**Comment by dragonattheend** [comment]
I'm getting the same warning: 

```
Warning:
Returning more (or less) than 1 row per `summarise()` group was deprecated in dplyr 1.1.0.
ℹ Please use `reframe()` instead.
ℹ When switching from `summarise()` to `reframe()`, remember that `reframe()` always returns an ungrouped data frame and adjust
  accordingly.
ℹ The deprecated feature was likely used in the tidyr package.
  Please report the issue at <https://github.com/tidyverse/tidyr/issues>.
---
Backtrace:
 1. ... %>% ungroup()
 4. tidyr:::complete.grouped_df(...)
 6. dplyr:::summarise.grouped_df(...)
 7. dplyr:::summarise_deprecate_variable_size()
 ``` 
from 

```
df %>% 
  group_by(id1, id2) %>% 
  complete(date = seq(min(date), max(date), by = "1 month")) %>% 
  ungroup()
```

**Comment by DavisVaughan** [comment]
@dragonattheend I think you probably need to update tidyr to v1.3.0

**Comment by dragonattheend** [comment]
> @dragonattheend I think you probably need to update tidyr to v1.3.0

Thanks! For some reason I didn't have it in the "Check for Package Updates". Now there's no warning.

**Comment by DavisVaughan** [comment]
Notes from Lionel make me think it would be somewhat difficult

> We previously had something in https://github.com/r-lib/rlang/blob/v1.0.5/R/compat-downstream-deps.R
I had to remove it because of the way it stored information in the search path to avoid repeatedly asking the user to upgrade. This was necessary for compatibility with RStudio Cloud where idle sessions are saved, and the only way of restoring the state I found was via the search path.
If that's needed we could revisit.
